device/cluster.cpp

```diff
@@ -478,6 +479,30 @@ void Cluster::add_chip(chip_id_t chip_id, std::unique_ptr<Chip> chip) {
     chips_.emplace(chip_id, std::move(chip));
 }
 
+uint32_t Cluster::get_tensix_harvesting_mask(
+    chip_id_t chip_id,
+    tt_ClusterDescriptor* cluster_desc,
+    bool perform_harvesting,
+    std::unordered_map<chip_id_t, uint32_t>& simulated_harvesting_masks) {
+    if (!perform_harvesting) {
+        log_info(LogSiliconDriver, "Skipping harvesting for chip {}.", chip_id);
+        return 0;
+    }
+    uint32_t tensix_harvesting_mask = cluster_desc->get_harvesting_info().at(chip_id);
+    uint32_t simulated_harvesting_mask = (simulated_harvesting_masks.find(chip_id) != simulated_harvesting_masks.end())
+                                             ? simulated_harvesting_masks.at(chip_id)
+                                             : 0;
+    if (simulated_harvesting_mask != 0) {
+        log_info(
+            LogSiliconDriver,
+            "Adding simulated harvesting mask {} for chip {} which has real harvesting mask {}.",
+            simulated_harvesting_mask,
+            chip_id,
+            tensix_harvesting_mask);
+    }
+    return tensix_harvesting_mask | simulated_harvesting_mask;
```

Comment:
I think we can't do this in this way. `tensix_harvesting_mask` is going to be retrieved from cluster descriptor, which is going to be based on the actual physical layout of the chip, and `simulated_harvesting_mask` is going to be in either logical (from 0 to N-1) or NOC0 (bit b is for the row with y=b on NOC). Either way masking this way won't work. I think there are 2 solutions here
- Make sure that simulated harvesting mask is passed based on physical chip layout. This probably won't be the solution since it is difficult to use
- Shuffle tensix_harvesting_mask to be in logical/NOC0 layout (figure out what is the layout of simulated mask) and the do the logical OR. I had similar change in #369, maybe we can do this shuffling out of Coordinate manager and soc descriptor
Let me know what you think

Reply:
I took this out of the class. The code is in CoordinateManager class, but it is a static function which is supposed to be called by the clients, currently called in Cluster. We discussed offline around this a bit. I wanted to avoid having two harvesting arguments in CoordinateManager which were used differently. This way. this api is simpler.